packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -31,34 +31,37 @@ export function attr_effect(dom, attribute, value) {
 }
 
 /**
- * @param {Element} dom
+ * @param {Element} element
  * @param {string} attribute
  * @param {string | null} value
  */
-export function attr(dom, attribute, value) {
+export function attr(element, attribute, value) {
 	value = value == null ? null : value + '';
 
-	if (DEV) {
-		check_src_in_dev_hydration(dom, attribute, value);
-	}
+	// @ts-expect-error
+	var attributes = (element.__attributes ??= {});
+
+	if (hydrating) {
+		attributes[attribute] = element.getAttribute(attribute);
+
+		if (attribute === 'src' || attribute === 'href' || attribute === 'srcset') {
```

Comment:
This is missing the DEV condition

Reply:
These attributes are skipped in dev and prod mode. The only difference is that in dev mode we additionally warn, which is what happens on the next line